Formula/u/uhdm.rb

```diff
@@ -0,0 +1,82 @@
+class Uhdm < Formula
+  include Language::Python::Virtualenv
+
+  desc "Universal Hardware Data Model, modeling of the SystemVerilog Object Model"
+  homepage "https://github.com/chipsalliance/UHDM"
+  url "https://github.com/chipsalliance/UHDM/archive/refs/tags/v1.71.tar.gz"
+  sha256 "3aa3783bc3fd5054bfe4347de1e2edfaa1bb3fbfbecdc3232413e826e95113c7"
+  license "Apache-2.0"
+  head "https://github.com/chipsalliance/UHDM.git", branch: "master"
+
+  depends_on "cmake" => :build
+  depends_on "python@3.11" => :build
+  depends_on "six" => :build
+  depends_on "pkg-config" => :test
+  depends_on "capnp"
+
+  resource "orderedmultidict" do
+    url "https://files.pythonhosted.org/packages/53/4e/3823a27d764bb8388711f4cb6f24e58453e92d6928f4163fdb01e3a3789f/orderedmultidict-1.0.1.tar.gz"
+    sha256 "04070bbb5e87291cc9bfa51df413677faf2141c73c61d2a5f7b26bea3cd882ad"
+  end
+
+  def install
+    venv = virtualenv_create(buildpath/"venv", "python3.11")
+    resources.each do |r|
+      venv.pip_install r
+    end
+    # Build shared library
+    system "cmake", "-S", ".", "-B", "build_shared",
+      "-DBUILD_SHARED_LIBS=ON",
+      "-DUHDM_BUILD_TESTS=OFF",
+      "-DUHDM_USE_HOST_GTEST=ON",
+      "-DUHDM_USE_HOST_CAPNP=ON",
+      "-DCMAKE_INSTALL_RPATH=#{rpath}",
+      "-DCMAKE_CXX_FLAGS=-fPIC",
+      "-DPython3_EXECUTABLE=#{buildpath}/venv/bin/python", *std_cmake_args
+    system "cmake", "--build", "build_shared"
+    system "cmake", "--install", "build_shared"
+    if OS.linux?
+      # Patch libuhdm.so location on linux, until fixed upstream
+      ln_s lib/"uhdm/libuhdm.so", lib
+    end
+  end
+
+  test do
+    ENV["PWD"] = testpath.to_s
+    system "test", "#{bin}/uhdm-cmp"
+    system "test", "#{bin}/uhdm-dump"
+    system "test", "#{bin}/uhdm-hier"
+    system "test", "#{bin}/uhdm-lint"
```

Comment:
These should be removed or the code should be adjusted to actually run the commands, currently it does nothing useful.
```suggestion
    assert_match "test", shell_output("#{bin}/uhdm-cmp")
    assert_match "some", shell_output("#{bin}/uhdm-dump"
    assert_match "output", shell_output("#{bin}/uhdm-hier")
    assert_match "here", shell_output("#{bin}/uhdm-lint")
```
or
```suggestion
```